### ISSUE TITLE
SC2094 complains when no pipes or writes are done.

### ISSUE THREAD

**Issue by smoser** [koalaman/shellcheck#1192]
#### For bugs
- Rule Id: SC2094
- My shellcheck version: 0.4.6
- [X] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh
line="" found_fn=""
for fname in "$@"; do
    # shellcheck disable=2162
    while read line; do
        case "$line" in
            foo) found_fn="$fname";;
        esac
    done <"$fname"
done
echo "$found_fn"

```

#### Here's what shellcheck currently says:
    $ shellcheck myscript
     
    Line 7:
                foo) found_fn="$fname";;
                              ^-- SC2094: Make sure not to read and write the same file in the same pipeline.
     
    Line 9:
        done <"$fname"
              ^-- SC2094: Make sure not to read and write the same file in the same pipeline.

#### Here's what I wanted or expected to see:
No errors, there are no pipelines here, and no writing to the file `$fname`

**Comment by koalaman** [comment]
Nice catch. Fixed by @ngzhian. Thanks!
